solr/src/main/java/com/yahoo/ycsb/db/SolrDBClient.java

```diff
@@ -0,0 +1,339 @@
+    // String collectionName = props.getProperty("table", DEFAULT_COLLECTION_NAME);
+    // TO-DO: Add a check that the collection exists
+    commitTime = Integer
+        .parseInt(props.getProperty("solr.commit.within.time", DEFAULT_COMMIT_WITHIN_TIME));
+    batchMode = Boolean.parseBoolean(props.getProperty("solr.batch.mode", DEFAULT_BATCH_MODE));
+
+    // Check if Solr cluster is running in SolrCloud or Stand-alone mode
+    Boolean cloudMode = Boolean.parseBoolean(props.getProperty("solr.cloud", DEFAULT_CLOUD_MODE));
+    System.out.println("Solr Cloud Mode = " + cloudMode);
+    if (cloudMode) {
+      System.out.println("Solr Zookeeper Remote Hosts = "
+          + props.getProperty("zookeeper.host.list", DEFAULT_REMOTE_ZOOKEEPER_HOST));
+      client = new CloudSolrClient(
+          props.getProperty("zookeeper.host.list", DEFAULT_REMOTE_ZOOKEEPER_HOST));
+    } else { 
+      client = new HttpSolrClient(
+          "http://" + props.getProperty("solr.host.list", DEFAULT_REMOTE_SOLR_HOST) + "/solr");
+    }
+  }
+
+  @Override
+  public void cleanup() throws DBException {
+    try {
+      client.close();
+    } catch (IOException e) {
+      e.printStackTrace();
+    }
+  }
+
+  /**
+   * Insert a record in the database. Any field/value pairs in the specified values HashMap will be
+   * written into the record with the specified record key.
+   *
+   * @param table
+   *          The name of the table
+   * @param key
+   *          The record key of the record to insert.
+   * @param values
+   *          A HashMap of field/value pairs to insert in the record
+   * @return Zero on success, a non-zero error code on error. See this class's description for a
+   *         discussion of error codes.
+   */
+  @Override
+  public Status insert(String table, String key, HashMap<String, ByteIterator> values) {
+    try {
+      SolrInputDocument doc = new SolrInputDocument();
+
+      doc.addField("id", key);
+      for (Entry<String, String> entry : StringByteIterator.getStringMap(values).entrySet()) {
+        doc.addField(entry.getKey(), entry.getValue());
+      }
+      UpdateResponse response;
+      if (batchMode) {
+        response = client.add(table, doc, commitTime);
+      } else {
+        response = client.add(table, doc);
+        client.commit(table);
+      }
+      return checkStatus(response.getStatus());
+
+    } catch (IOException | SolrServerException e) {
+      e.printStackTrace();
+    }
+    return Status.ERROR;
+  }
+
+  /**
+   * Delete a record from the database.
+   *
+   * @param table
+   *          The name of the table
+   * @param key
+   *          The record key of the record to delete.
+   * @return Zero on success, a non-zero error code on error. See this class's description for a
+   *         discussion of error codes.
+   */
+  @Override
+  public Status delete(String table, String key) {
+    try {
+      UpdateResponse response = client.deleteById(table, key, commitTime);
```

Comment:
commitWithin here but not an if statement check and explicit commit like other places?